samples/cpp/ela.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+  @file ela.cpp
+  @
+  @brief ELA allows to see visually the changes made in a JPG image based in it's compression error analysis. Based in Eliezer Bernart example.
+  @date Jun 24, 2018
+*/
+
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <iostream>
+#include <vector> 
+
+const char* keys =
+    "{ help h      | | Print help message. }"
+    "{ input i     | | Input image to calc ELA algorithm. }";
+
+
+using namespace cv;
+
+int scale_value = 7;
+int quality = 95;
+Mat image;
+Mat compressed_img;
+
+static void processImage(int , void* )
+{
+    // Compression jpeg
+    std::vector<int> compressing_factor;
+    compressing_factor.push_back(CV_IMWRITE_JPEG_QUALITY);
+    compressing_factor.push_back(quality);
+
+    imwrite("temp.jpg", image, compressing_factor);
```

Comment:
do not write compressed image to a file, use `encode` and `decode` functions from imgcodecs.